resources/osx/preinstall-extras

```diff
@@ -0,0 +1,16 @@
+## This works around an issue with the OSX service names in
+## puppet-agent. We're using a short service name intead of
+## the fully qualified name (puppet instead of com.puppetlabs.puppet)
+
+## Putting this workaround here until we can fix the service names in
+## the next major version, at which point the more generic logic in
+## vanagon will do the right thing
+
+if [ -n "$foundpkg" ]; then
```

Comment:
Do we care if `the length of STRING is nonzero` (`-n`) or `the length of STRING is zero` (`-z`)?